src/sema/unused_variable.rs

```diff
@@ -43,6 +49,49 @@ pub fn assigned_variable(ns: &mut Namespace, exp: &Expression, symtable: &mut Sy
     }
 }
 
+// We have two cases here
+//  contract c {
+//      int[2] case1;
+//
+//      function f(int[2] storage case2) {
+//          case1[0] = 1;
+//          case2[0] = 1;
+//      }
+//  }
+//  The subscript for case1 is an assignment
+//  The subscript for case2 is a read
```

Comment:
Isn't `case2` an assignment too?

Reply:
An assignment to `case2` would look like:
```solidity
  case2 = ...;
```
Another way to think about this:

```solidity
contract test {
     function bar(int[] storage foo) internal view {
                foo[0] = 102;
  }
}
```
If you compile this with current `main`, you get a warning saying parameter `foo` is unused. It's clearly not